### PR TITLE
Improve Chromecast on different subnet docs

### PR DESCRIPTION
**Pull request by slovdahl**
## Proposed change

Add information about the shortcomings identified in https://github.com/home-assistant/core/issues/27174.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/66974

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by emontnemery — COMMENTED** [review]
Can you confirm the address translation is to make the mDNS based discovery work, or to make TCP-based communication with the chromecasts work?

File: source/_integrations/cast.markdown [hunk 1]
```diff
@@ -349,7 +349,7 @@ The Google Cast integration relies on the [Zeroconf integration](/integrations/z
 
 Cast devices can only be automatically discovered if they are on the same subnet as Home Assistant because mDNS packets are not routed across subnets.
 Automatic discovery in setups with cast devices on a different subnet than Home Assistant is not recommended and not supported.
-If it is not possible, it's necessary to either enable mDNS forwarding between the subnets or to configure a list of known hosts.
+If it is not possible, it's necessary to either enable mDNS forwarding between the subnets or to configure a list of known hosts. In addition to that, source NAT may need to be enabled to make requests from Home Assistant to the Chromecast appear as coming from the same subnet.
```
Comment (emontnemery) [inline source/_integrations/cast.markdown]:
```suggestion
If it is not possible to have Home Assistant and the cast devices on the same subnet, it's necessary to either enable mDNS forwarding between the subnets or to configure a list of known hosts.

In addition to the problem with mDNS discovery across subnets, some Cast devices have been reported to block TCP based traffic not originating from their own subnet, if this is the case source NAT needs to be enabled to make TCP based traffic from Home Assistant to the Chromecast appear as originating from the same subnet.

Setups with cast devices on a different subnet than Home Assistant are strongly advised against and not supported.
```

File: source/_integrations/cast.markdown [hunk 2]
```diff
@@ -349,7 +349,7 @@ The Google Cast integration relies on the [Zeroconf integration](/integrations/z
 
 Cast devices can only be automatically discovered if they are on the same subnet as Home Assistant because mDNS packets are not routed across subnets.
 Automatic discovery in setups with cast devices on a different subnet than Home Assistant is not recommended and not supported.
```
Comment (emontnemery) [inline source/_integrations/cast.markdown]:
```suggestion
```

**Review by emontnemery — CHANGES_REQUESTED** [review]

**Comment by emontnemery** [comment]
Rephrased assuming source NAT is required to make TCP based requests work.
@slovdahl Can you please help confirm this is the case?

**Comment by emontnemery** [comment]
I'm setting this PR to draft until we have a confirmation from @slovdahl

**Comment by slovdahl** [comment]
Sorry for the delay.

> Rephrased assuming source NAT is required to make TCP based requests work.
@slovdahl Can you please help confirm this is the case?

Yes, that was the intention at least. mDNS discovery has always worked fine for me, but I've had some problems with getting e.g. the playing/paused state synchronized in a timely manner to HA, and initially it seemed like enabling source NAT improved the situation. But in the end, I'm still having the same kind of problems, so it's probably something else.

File: source/_integrations/cast.markdown (hunk 1 above)
Comment (slovdahl) [reply source/_integrations/cast.markdown]: Thanks, good suggestion 👍

**Comment by emontnemery** [comment]
> initially it seemed like enabling source NAT improved the situation. But in the end, I'm still having the same kind of problems, so it's probably something else

Should we close the PR then if you're not sure enabling source NAT makes any difference?

**Comment by slovdahl** [comment]
No strong opinion either way as far as I'm concerned. It _did_ seem to do the trick for @nayfield according to #66974 though.

**Pull request closed without merging**